specs/WebNotification.md

```diff
@@ -0,0 +1,747 @@
+  /// ICoreWebView2UnsignedLongCollection.
+  [propget] HRESULT Count([out, retval] UINT* value);
+
+  /// Gets the unsigned long integer at the given index.
+  HRESULT GetValueAtIndex([in] UINT index, [out, retval] UINT64* value);
+}
+
+/// The caller implements this interface to receive the result of reporting a notification clicked.
+[uuid(1013C0D5-5F0C-4BF8-BA52-9D76AFA20E83), object, pointer_default(unique)]
+interface ICoreWebView2NotificationReportClickedCompletedHandler : IUnknown {
+    HRESULT Invoke([in] HRESULT errorCode);
+}
+
+/// The caller implements this interface to receive the result of reporting a notification closed.
+[uuid(D8B63F74-1D78-4B4C-ACA9-7CB63FDFD74C), object, pointer_default(unique)]
+interface ICoreWebView2NotificationReportClosedCompletedHandler : IUnknown {
+    HRESULT Invoke([in] HRESULT errorCode);
+}
+
+/// This is the ICoreWebView2Notification that represents a [HTML Notification
+/// object](https://developer.mozilla.org/docs/Web/API/Notification).
+[uuid(E3F43572-2930-42EB-BD90-CAC16DE6D942), object, pointer_default(unique)]
+interface ICoreWebView2Notification : IUnknown {
+  /// Add an event handler for the `CloseRequested` event.
+  /// This event is raised when the notification is closed by the web code, such as 
+  /// through `notification.close()`.
+  HRESULT add_CloseRequested(
+      [in] ICoreWebView2NotificationCloseRequestedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_CloseRequested`.
+  HRESULT remove_CloseRequested(
+      [in] EventRegistrationToken token);
+
+  /// The host may run this to report the notification has been displayed and it
+  /// will cause the [show](https://developer.mozilla.org/docs/Web/API/Notification/show_event)
+  /// event to be raised for non-persistent notifications.
+  /// You should only run this if you are handling the `NotificationReceived`
+  /// event. Returns `E_ABORT` if `Handled` is `FALSE` when this is called.
+  HRESULT ReportShown();
+
+  /// The host may run this to report the notification has been clicked, and it
+  /// will cause the
+  /// [click](https://developer.mozilla.org/docs/Web/API/Notification/click_event)
+  /// event to be raised for non-persistent notifications and the
+  /// [notificationclick](https://developer.mozilla.org/docs/Web/API/ServiceWorkerGlobalScope/notificationclick_event)
+  /// event for persistent notifications. Use `ReportClickedWithAction` to specify an
+  /// action to activate a persistent notification.
+  /// You should only run this if you are handling the `NotificationReceived`
+  /// event. Returns `E_ABORT` if `Handled` is `FALSE` or `ReportShown` has not
+  /// been run when this is called.
+  HRESULT ReportClicked([in] ICoreWebView2NotificationReportClickedCompletedHandler* handler);
+
+  /// The host may run this to report the persistent notification has been
+  /// activated with a given action, and it will cause the
```

Comment:
There is no sample showing "with an action" vs ReportClicked.  Does no-action mean customer clicked the notification body?